server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -137,6 +138,13 @@ public class TransportSearchAction extends HandledTransportAction<SearchRequest,
         Property.NodeScope
     );
 
+    public static final Setting<Boolean> SEARCH_QUERY_CATEGORIZATION_ENABLED_SETTING = Setting.boolSetting(
```

Comment:
Can we call it metrics to be used generically for all query metrics?

```suggestion
    public static final Setting<Boolean> SEARCH_QUERY_METRICS_ENABLED_SETTING = Setting.boolSetting(
```